src/datumaro/plugins/data_formats/voc/base.py

```diff
@@ -52,21 +53,31 @@ def __init__(
         task: Optional[VocTask] = VocTask.voc,
         *,
         subset: Optional[str] = None,
+        voc_importer_type: VocImporterType = VocImporterType.default,
         ctx: Optional[ImportContext] = None,
     ):
-        if not osp.isfile(path):
-            raise DatasetImportError(f"Can't find txt subset list file at '{path}'")
-        self._path = path
-        self._dataset_dir = osp.dirname(osp.dirname(osp.dirname(path)))
-
-        self._task = task
-
         if not subset:
             subset = osp.splitext(osp.basename(path))[0]
 
         super().__init__(subset=subset, ctx=ctx)
 
-        self._categories = self._load_categories(self._dataset_dir)
+        if voc_importer_type == VocImporterType.default:
+            dataset_dir = osp.dirname(osp.dirname(osp.dirname(path)))
+            self._image_dir = osp.join(dataset_dir, VocPath.IMAGES_DIR)
+            self._anno_dir = osp.join(dataset_dir, VocPath.ANNOTATIONS_DIR)
+            self._mask_dir = osp.join(dataset_dir, VocPath.SEGMENTATION_DIR)
+            self._inst_dir = osp.join(dataset_dir, VocPath.INSTANCES_DIR)
+        elif voc_importer_type == VocImporterType.roboflow:
+            dataset_dir = path
+            self._image_dir = dataset_dir
+            self._anno_dir = dataset_dir
```

Comment:
Is it ok no definition for `self._mask_dir` and `self._inst_dir` if `voc_importer_type == VocImporterType.roboflow`?

Reply:
Roboflow Segmentation only supports COCO JSON format as shown below.
![image](https://github.com/openvinotoolkit/datumaro/assets/89109581/25ecfaaf-c092-4353-ad8d-9443552893ae)